### DISCUSSION TITLE
is file-based routing with extending config possible?

### DISCUSSION THREAD

**Discussion by itsmnthn** [nitrojs/nitro#705] [Q&A] [answered]
So the downside of file-based routing is to change the route I need to rename files/directories and this is not good for GitHub history or managing multiple APIs that change in versions due to a lot of changes.

I have a project that uses static file storage to store static data and adding new data requires the API version to change. Because nitro uses file base route I defined `routes/v3.1/...` and this v3.1 needs to update. and hence renaming dir is not a good option every time. And, this same goes for the Nuxt project too.

So Is there a way to configure routing prefixes?

**Comment by pi0** [comment, marked as answer]
Hi. You can use a placeholder param name like `routes/[version]/test.ts`.
